### DISCUSSION TITLE
Tomo3D and cryoCARE integration

### DISCUSSION THREAD

**Discussion by xingzhe1129** [nextpyp/.github#36] [Ideas]
It seems that tomo3d and cryoCARE softwares work pretty well. Could you please try to integrate these two softwares into PYP? Thank you so much.

**Comment by abartesaghi** [comment]
Can you provide more details about what these packages do and any links to the software, specific versions and their licenses?

**Reply by xingzhe1129** [reply]
tomo3d (2.2 version, https://sites.google.com/site/3demimageprocessing/tomo3d) implements a multithreaded vectorized approach to tomographic reconstruction that takes full advantage of the power in modern multicore computers. Version 2.2 contains implementations of the well-known WBP and SIRT methods, and WBP using a SIRT-like filter (also known as 'Fake SIRT') as well as CTF deconvolution using D. Tegunov's filter.

cryoCARE_pip (https://github.com/juglab/cryoCARE_pip) is a memory efficient implementation of cryoCARE (https://github.com/juglab/cryoCARE_T2T).This setup trains a denoising U-Net for tomographic reconstruction according to the Noise2Noise training paradigm.

**Reply by abartesaghi** [reply]
OK, I couldn't immediately find information about licenses for tomo3d but we will look into it. Tomograms reconstructed by nextPYP are usually heavily binned and already very fast to compute, but I assume there could be cases where users may want to build unbinned tomograms and performance improvements from tomo3d would be desirable there. Regarding denoising, nextPYP already supports Topaz's N2N-based 3D denosing. Are there any additional features provided by cryoCARE that would justify integrating it to nextPYP?

**Reply by xingzhe1129** [reply]
No, cryoCARE is also used for denosing but I see people said that cryoCARE works better than topaz sometimes. Thanks.
